### PR TITLE
Fix p3_stand_alone and p3_shoc_coupled unit tests.

### PR DESCRIPTION
**Pull request by AaronDonahue**
This commit creates a new atmospheric process for initialization of
microphysics variables, SAMicrophysics, referring to
"stand-alone-microphysics".  It is a temporary fix to handle an issue
that arises from running p3 stand alone which requires variables declared
as "required" or "input" to be somehow initialized without calling other
processes.

The fix is to include a stand-alone process which treats all variables as
computed, thus allowing for the changes to the variable to be accessible to
the main p3 microphysics call.  The stand-alone process only has an init call
defined since it isn't expected to do anything during the run and finalize
sequence.

This commit also removes an old compiler flag which did not build the p3
and p3_shoc stand-alone tests if SCREAM_FPE was on.

[BFB]*

*Although there are now two new tests in the debug_nopack_fpe suite owing
to p3 and p3_shoc tests now being turned on.

### CODE REVIEW

File: components/scream/src/physics/p3/scream_p3_interface.F90 [hunk 1]
```diff
@@ -64,6 +64,32 @@ subroutine p3_init_f90 (q,T,zi,pmid,pdel,ast,naai,npccn) bind(c)
     integer(kind=c_int) :: i, k
     logical(kind=c_bool) :: masterproc
 
+    call p3_init(micro_p3_lookup_dir,micro_p3_tableversion)
```
Comment (bartgol) [inline components/scream/src/physics/p3/scream_p3_interface.F90]: Looks like p3_init_f90 is no longer using any of its inputs. If so, we should change its signature (avoid unused stuff).

File: components/scream/src/physics/p3/atmosphere_microphysics.hpp [hunk 2]
```diff
@@ -83,6 +83,79 @@ class P3Microphysics : public scream::AtmosphereProcess
...
 
+/*
+ * This class is microphysics stub meant to be used for initialization of
+ * stand-alone tests.  It is similar to P3Microphysics except that it won't
+ * actually calculate anything, it will only initialize the input fields needed
+ * by atmophseric microphysics.
+ * TODO: Switch to a general stand-alone class for initialization of any stand-alone
+ * test.
+*/
+class SAMicrophysics : public scream::AtmosphereProcess
```
Comment (bartgol) [inline components/scream/src/physics/p3/atmosphere_microphysics.hpp]: De gustibus, but you could rename the class to something more meaningfull, like `P3InputsProvider` or something like that. This process, per se, does not implement microphysics, but rather simply makes sure P3 inputs are valid.

File: components/scream/src/physics/p3/scream_p3_interface.F90 (hunk 1 above)
Comment (AaronDonahue) [reply components/scream/src/physics/p3/scream_p3_interface.F90]: oh yeah, for sure.  I'll make the change and test.

File: components/scream/src/physics/p3/atmosphere_microphysics.hpp (hunk 2 above)
Comment (AaronDonahue) [reply components/scream/src/physics/p3/atmosphere_microphysics.hpp]: Yes, that could certainly work.  I'll make the change.

**Comment by AaronDonahue** [comment]
Changes have been made and the tests appear to pass.  Side-question.  The AutoTester appears to be not working because of the timeout issue.  Would it be alright for me to simply test this PR offline using the gather-all-data command and if it passes go ahead and merge with AutoTester approval?

**Comment by bartgol** [comment]
Re: Autotester. Yes, the AT is simply a tool to save people some time/typing. Unfortunately, there's some scheduling issue on our GPU machine on the open network, which cause the jobs to stall. If you can manually test a CPU and a GPU machine, and they both pass, this can be merged.

**Comment by AaronDonahue** [comment]
great, I'll do that

**Comment by AaronDonahue** [comment]
Testing on CPU-quartz and GPU-lassen machines at LC passes for all tests.  I'm going to merge.

**Pull request merged**